docs/en/latest/http3.md

```diff
@@ -0,0 +1,188 @@
+```
+
+Next, generate the key and certificate with a common name for APISIX, and sign with the CA certificate:
+
+```shell
+openssl genrsa -out server.key 2048 && \
+  openssl req -new -sha256 -key server.key -out server.csr -subj "/CN=test.com" && \
+  openssl x509 -req -days 36500 -sha256 -extensions v3_req \
+  -CA ca.crt -CAkey ca.key -CAserial ca.srl -CAcreateserial \
+  -in server.csr -out server.crt
+```
+
+### Configure HTTPS
+
+Optionally load the content stored in `server.crt` and `server.key` into shell variables:
+
+```shell
+server_cert=$(cat server.crt)
+server_key=$(cat server.key)
+```
+
+Create an SSL certificate object to save the server certificate and its key:
+
+```shell
+curl -i "http://127.0.0.1:9180/apisix/admin/ssls" -X PUT -d '
+{
+  "id": "quickstart-tls-client-ssl",
+  "sni": "test.com",
+  "cert": "'"${server_cert}"'",
+  "key": "'"${server_key}"'"
+}'
+```
+
+### Create a Route
+
+Create a sample route to `httpbin.org`:
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT -d '
+{
+  "id":"httpbin-route",
+  "uri":"/get",
+  "upstream": {
+    "type":"roundrobin",
+    "nodes": {
+      "httpbin.org:80": 1
+    }
+  }
+}'
+```
+
+### Verify HTTP/3 Connections
+
+Install [curl `>= 7.88.0`](https://curl.se/changes.html#7_88_0) for HTTP/3 support, or other curl that have HTTP/3 support.
+
+Send a request to the route:
+
+```shell
+curl -kv --http3-only \
+  -H "Host: test.com" \
+  -H "content-length: 0" \
+  --resolve "test.com:9443:127.0.0.1" "https://test.com:9443/get"
+```
+
+You should receive an `HTTP/3 200` response similar to the following:
+
+```text
+* Added test.com:9443:127.0.0.1 to DNS cache
+* Hostname test.com was found in DNS cache
+*   Trying 127.0.0.1:9443...
+* QUIC cipher selection: TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_CCM_SHA256
+* Skipped certificate verification
+* Connected to test.com (127.0.0.1) port 9443
+* using HTTP/3
+* [HTTP/3] [0] OPENED stream for https://test.com:9443/get
+* [HTTP/3] [0] [:method: GET]
+* [HTTP/3] [0] [:scheme: https]
+* [HTTP/3] [0] [:authority: test.com]
+* [HTTP/3] [0] [:path: /get]
+* [HTTP/3] [0] [user-agent: curl/8.7.1]
+* [HTTP/3] [0] [accept: */*]
+> GET /get HTTP/3
+> Host: test.com
+> User-Agent: curl/8.7.1
+> Accept: */*
+>
+* Request completely sent off
+< HTTP/3 200
+...
+{
+  "args": {},
+  "headers": {
+    "Accept": "*/*",
+    "Content-Length": "0",
+    "Host": "test.com",
+    "User-Agent": "curl/8.7.1",
+    "X-Amzn-Trace-Id": "Root=1-6656013a-27da6b6a34d98e3e79baaf5b",
+    "X-Forwarded-Host": "test.com"
+  },
+  "origin": "172.19.0.1, 123.40.79.456",
+  "url": "http://test.com/get"
+}
+* Connection #0 to host test.com left intact
+```
+
+## Known Issues
+
+- For APISIX-3.9, test cases of Tongsuo will fail because the Tongsuo does not support QUIC TLS.
+- For HTTP/2 or HTTP/3 the request should send `content-length` header no matter what HTTP methods you use.
```

Comment:
The `known issues` section was added by the original author of PR #10990 where it suggests HTTP/2 and HTTP/3 should both send `content-length: 0`, e.g.

![image](https://github.com/apache/apisix/assets/39619599/5ce0a89e-6148-47e3-af81-ea658e58e076)

In trying to understand why it is needed, I removed the header and noticed the connection was still successful:

![xxU9QMQW1X](https://github.com/api7/docs/assets/39619599/8765cd8c-9ffa-4bd4-a45b-5b744202932b)

If anyone understands better, please share your knowledge.

Reply:
I think this sentence can be removed, I haven't found any relevant standard documentation to support this point.

Reply:
The test cases have `content-length: 0` (but it's using a different curl). I'm running cI tests in my repo to see whether they are still needed.